elastic/beats
# [Heartbeat] Add service_name option for APM integration

## What does this PR do?

Adds a new standard `service_name` option to the heartbeat config file. While possible with `fields` already, adding this as a first class option encourages use of this important field for integration.

First step toward https://github.com/elastic/uptime/issues/220

This PR also refactors some internal bits where we were passing too many parameters already, and adding `service_name` would just be too much. We now pass a single larger struct for common monitor options which cleans up a lot of the code.

## Checklist



## How to test this PR locally

Run against ES with `service_name` in your config and see that `service.name` is populated.
